pkg/errorhandling/errorhandling.go

```diff
@@ -1,14 +1,26 @@
 package errorhandling
 
 import (
+	"fmt"
 	"os"
 	"strings"
+	"unsafe"
 
 	"github.com/hashicorp/go-multierror"
+	jsoniter "github.com/json-iterator/go"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+func init() {
+	jsoniter.RegisterTypeDecoderFunc("[]error", UnmarshalErrorSliceJSON)
+	jsoniter.RegisterTypeDecoderFunc("error", UnmarshalErrorJSON)
+	jsoniter.RegisterTypeEncoderFunc("[]error", MarshalErrorSliceJSON, MarshalErrorSliceJSONIsEmpty)
+	jsoniter.RegisterTypeEncoderFunc("error", MarshalErrorJSON, MarshalErrorJSONIsEmpty)
```

Comment:
By using this we also force this on all bindings users. I guess we want this but pkg/bindings user might get unexpected results when the also use `github.com/json-iterator/go` in their code. Unfortunately it looks like they only support global encoders/decoders https://github.com/json-iterator/go/blob/024077e996b048517130b21ea6bf12aa23055d3d/reflect_extension.go#L13-L16 so I don't see a way around this.

Can we document this somewhere in pkg/bindings?

Reply:
I will add that in a README.